test/src/folding/parsers/python_parser_test.dart

```diff
@@ -0,0 +1,207 @@
+    test('parses spaces', () {
+      const examples = [
+        _Example(
+          'Python. Empty string',
+          code: '',
+          expected: [],
+        ),
+        //
+        _Example(
+          'Python. Nesting with multiline list',
+          code: '''
+class Mapping:
+    def __init__(self, iterable):
+        self.items_list = []
+        self.__update(iterable)
+
+    def update(self, iterable):
+        for item in iterable:
+            self.items_list.append(item)
+
+        a = [
+            5,
+            6,
+            7,
+            8
+        ]''',
+          expected: [
+            _FB(startLine: 0, endLine: 14, type: FoldableBlockType.indent),
+            _FB(startLine: 1, endLine: 3, type: FoldableBlockType.indent),
+            _FB(startLine: 5, endLine: 14, type: FoldableBlockType.indent),
+            _FB(startLine: 6, endLine: 7, type: FoldableBlockType.indent),
+            _FB(startLine: 9, endLine: 14, type: FoldableBlockType.brackets),
+          ],
+        ),
+        //
+        _Example(
+          'Python. Invalid code',
+          code: '''
+class Mapping:
+    def __init__(self, iterable)
+        self.items_list = []
+        self.__update(iterable)
+
+    def update(self, iterable)
+        for item in iterable
+            self.items_list.append(item)
+
+        a = [
+            5,
+            6,
+            7,
+            8
+        ]''',
+          expected: [
+            _FB(startLine: 0, endLine: 14, type: FoldableBlockType.indent),
+            _FB(startLine: 1, endLine: 3, type: FoldableBlockType.indent),
+            _FB(startLine: 5, endLine: 14, type: FoldableBlockType.indent),
+            _FB(startLine: 6, endLine: 7, type: FoldableBlockType.indent),
+            _FB(startLine: 9, endLine: 14, type: FoldableBlockType.brackets),
+          ],
+        ),
+        //
+        _Example(
+          'Python. Nesting list',
+          code: '''
+class Mapping:
+    def __init__(self, iterable):
+        a = [
+            [
+                5,
+                6
+            ],
+            [
+                7,
+                8
+            ]
+        ]''',
+          expected: [
+            _FB(startLine: 0, endLine: 11, type: FoldableBlockType.indent),
+            _FB(startLine: 1, endLine: 11, type: FoldableBlockType.indent),
+            _FB(startLine: 2, endLine: 11, type: FoldableBlockType.brackets),
+            _FB(startLine: 3, endLine: 6, type: FoldableBlockType.brackets),
+            _FB(startLine: 7, endLine: 10, type: FoldableBlockType.brackets),
+          ],
+        ),
+        //
+        _Example(
+          'Python. Single-line comments',
+          code: '''
+class Mapping:
+    def __init__(self, iterable):
+        a = [
+            [
+              # comment
+              # another comment
+              # third comment
+                5,
+                6
+            ],
+            [
+                7,
+                8
+            ]
```

Comment:
Please add another block with comments block indented further than the list content.
Please add a lone single line comment to make sure it does not form a block.